src/features/harborList/utils.ts

```diff
@@ -67,3 +68,25 @@ export const getHarborsData = (data: HARBORS | undefined) => {
   }
   return [];
 };
+
+export const calculateBerthSummary = (
+  data: { numberOfPlaces: number; numberOfFreePlaces: number }[]
+): BerthSummaryProps => {
+  if (data.length === 0) {
+    return {};
+  }
+
+  const berthCount = data.reduce<number>((acc, harbor) => {
+    return acc + harbor.numberOfPlaces;
+  }, 0);
+
+  const freeCount = data.reduce<number>((acc, harbor) => {
+    return acc + harbor.numberOfFreePlaces;
+  }, 0);
```

Comment:
Can't we perform this calculation by looping through the data only once?

Something like:
```suggestion
  return data.reduce((acc, harbor) => {
    const berthCount = acc.berthCount + harbor.numberOfPlaces;
    const freeCount = acc.freeCount + harbor.numberOfFreePlaces;
    return {
      berthCount,
      freeCount,
      reservedCount: berthCount - freeCount,
  }
  }, {
    berthCount: 0,
    freeCount: 0,
    reservedCount: 0,
  });
```

Reply:
Indeed we can, thanks!